arithmetization/src/main/java/net/consensys/linea/extradata/LineaExtraDataHandler.java

```diff
@@ -0,0 +1,145 @@
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import lombok.extern.slf4j.Slf4j;
+import net.consensys.linea.config.LineaProfitabilityConfiguration;
+import org.apache.commons.lang3.mutable.MutableLong;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt32;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.datatypes.rpc.JsonRpcResponseType;
+import org.hyperledger.besu.plugin.data.AddedBlockContext;
+import org.hyperledger.besu.plugin.data.BlockHeader;
+import org.hyperledger.besu.plugin.services.BesuEvents;
+import org.hyperledger.besu.plugin.services.BlockchainService;
+import org.hyperledger.besu.plugin.services.RpcEndpointService;
+
+@Slf4j
+public class LineaExtraDataHandler implements BesuEvents.BlockAddedListener {
+  private final RpcEndpointService rpcEndpointService;
+  private final ExtraDataParser[] extraDataParsers;
+
+  public LineaExtraDataHandler(
+      final RpcEndpointService rpcEndpointService,
+      final BlockchainService blockchainService,
+      final LineaProfitabilityConfiguration profitabilityConf) {
+    this.rpcEndpointService = rpcEndpointService;
+    extraDataParsers = new ExtraDataParser[] {new Version1Parser(profitabilityConf)};
+    onStartup(blockchainService);
+  }
+
+  private void onStartup(final BlockchainService blockchainService) {
+    consumeExtraData(blockchainService.getChainHeadHeader());
+  }
+
+  @Override
+  public void onBlockAdded(final AddedBlockContext addedBlockContext) {
+    consumeExtraData(addedBlockContext.getBlockHeader());
+  }
+
+  private void consumeExtraData(final BlockHeader blockHeader) {
+    final var rawExtraData = blockHeader.getExtraData();
+
+    if (!Bytes.EMPTY.equals(rawExtraData)) {
+      for (final ExtraDataParser extraDataParser : extraDataParsers) {
+        if (extraDataParser.canParse(rawExtraData)) {
+          final var extraData = rawExtraData.slice(1);
+          extraDataParser.parse(extraData);
+          return;
+        }
+      }
+      log.warn("unsupported extra data field {}", rawExtraData.toHexString());
+    }
+  }
+
+  private interface ExtraDataParser {
+    boolean canParse(Bytes extraData);
+
+    void parse(Bytes extraData);
+
+    static Long toLong(final Bytes fieldBytes) {
+      return UInt32.fromBytes(fieldBytes).toLong();
+    }
+  }
+
+  @SuppressWarnings("rawtypes")
+  private class Version1Parser implements ExtraDataParser {
+    private final LineaProfitabilityConfiguration profitabilityConf;
+    private final FieldConsumer[] fieldsSequence;
+    private final MutableLong currFixedCostKWei = new MutableLong();
+    private final MutableLong currVariableCostKWei = new MutableLong();
+
+    public Version1Parser(final LineaProfitabilityConfiguration profitabilityConf) {
+      this.profitabilityConf = profitabilityConf;
+
+      final FieldConsumer fixedGasCostField =
```

Comment:
Sorry it's coming late in the game, but only after all the changes I got this idea. We can isolate this wei vs kwei complexity here by parsing kwei values from extraData, but setting wei everywhere else. This way we'd get rid of this import of `net.consensys.linea.config.LineaProfitabilityConfiguration.WEI_IN_KWEI` in `net.consensys.linea.extradata` package, which seems wrong and make profitability logic simpler by not having to care about different measurement units

Reply:
I am fine using always wei, since there is no risk of overflow having that kwei is only 4bytes.
Will also change these parameters to be wei
`plugin-linea-fixed-gas-cost-wei`
`plugin-linea-variable-gas-cost-wei`